src/lib/action.ts

```diff
@@ -213,19 +225,51 @@ export default class ActionHandler {
     }
   };
 
-  requestTeX = async (): Promise<void> => {
+  requestTeX = async (): Promise<
+    "success" | "no latex entered" | "invalid latex"
+  > => {
     const text = window.prompt("Enter LaTeX source");
```

Comment:
nit: now that i think about it this should probably have a colon at the end xp

Reply:
This is against what I'd usually do (the user already knows that we're prompting them, even more so with form input labels where the distinction is less clear) but i don't think it significantly hurts so i'll change it if you still want